### PR TITLE
Grw 1311/fix/update storybook fix build warnings

### PR DESCRIPTION
**Pull request by alebedev**
## What?

Fix many pages of storybook build warnings by updating babel and storybook

## Why?

Removing warnings improve developer experience, we don't want warnings that we ignore

**Ticket(s): [GRW-1311]**

[GRW-1311]: https://hedvig.atlassian.net/browse/GRW-1311?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -193,6 +192,7 @@
   },
   "sideEffects": false,
   "resolutions": {
-    "koa-router/path-to-regexp": "6.1.0"
+    "koa-router/path-to-regexp": "6.1.0",
+    "@types/react": "16.14.29"
```
Comment (robinandeer) [inline package.json:196]: I'm not 100% how this works but do we need this or can we pin the version in the dependencies above?

**Review by robinandeer — APPROVED** [review]
Great stuff!

[cathack](https://app.graphite.dev/macros/HedvigInsurance)

**Review by guilhermespopolin — APPROVED** [review]
Nice 👷🏻

File: package.json (hunk 1 above)
Comment (alebedev) [reply package.json:196]: This forces any direct or transitive dependency to use specific version of package (see https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/)

Using resolutions is usually last resort for conflicting transitive dependencies.  This is exactly our case - storybook and friends all want @types/react ^18.something and we don't want to go back to very old storybook that wants react 16

Comment (robinandeer) [reply package.json:196]: great, thanks :)

**Review by robinandeer — APPROVED** [review]

**Pull request merged**